pkg/dockerregistry/server/quotarestrictedblobstore.go

```diff
@@ -72,5 +80,44 @@ type quotaRestrictedBlobWriter struct {
 func (bw *quotaRestrictedBlobWriter) Commit(ctx context.Context, provisional distribution.Descriptor) (canonical distribution.Descriptor, err error) {
 	context.GetLogger(ctx).Debug("(*quotaRestrictedBlobWriter).Commit: starting")
 
+	if err := admitBlobWrite(ctx, bw.repo, provisional.Size); err != nil {
+		context.GetLogger(ctx).Error(err.Error())
+		return distribution.Descriptor{}, err
+	}
+
 	return bw.BlobWriter.Commit(ctx, provisional)
 }
+
+// admitBlobWrite checks whether the blob does not exceed image limit ranges if set. Returns ErrAccessDenied
+// error if the quota is exceeded.
+func admitBlobWrite(ctx context.Context, repo *repository, size int64) error {
+	if err := admitLimitRanges(ctx, repo, size); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
If this method only calls the next one, why not just get rid of it? Both have exactly the same signatures. Maybe previously it was necessary, but not anymore.